### ISSUE TITLE
Bug with @VirtualColumn not being returned correctly

### ISSUE THREAD

**Issue by macleysousa** [typeorm/typeorm#10638]
### Issue description

@VirtualColumn not being returned correctly

### Expected Behavior

``` js
@Entity({ name: 'integrations' })
export class IntegrationEntity {
  @Exclude()
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ApiProperty()
  @PrimaryColumn({ name: 'client_id', type: 'uuid' })
  client_id: string;

  @ApiProperty()
  @VirtualColumn({ query: (alias) => `select "DMS" from config_client where "client_id"=${alias}.client_id` })
  dms: string;
}

```

It was expected that @VirtualColumn would return the correct values according to the logic defined in the decorator.

``` json
{
  "client_id": "1b97a2d2-91b8-4dab-9593-906da6d95d36",
  "dms": "NBS"
}
```

### Actual Behavior

``` js
@Entity({ name: 'integrations' })
export class IntegrationEntity {
  @Exclude()
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ApiProperty()
  @PrimaryColumn({ name: 'client_id', type: 'uuid' })
  client_id: string;

  @ApiProperty()
  @VirtualColumn({ query: (alias) => `select "DMS" from config_client where "client_id"=${alias}.client_id` })
  dms: string;
}

```

DNS column is not being returned

``` json
{
  "client_id": "1b97a2d2-91b8-4dab-9593-906da6d95d36"
}
```

### Steps to reproduce

1. Add the @VirtualColumn decorator to an entity or class.
2. Define the expected logic for the virtual value.
3. Observe the obtained result.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  windows 10       |
| Node.js version     |  18.18.0  |
| Typescript version  |  5.3.3  |
| TypeORM version     | 0.3.19  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, and I know how to start.

**Comment by jorenvandeweyer** [comment]
How do you select the virtual column, can you give an example of the implementation?

**Comment by kevinwilson541** [comment]
As an example, usage of `findOne` or `findOneOrFail`, without the `select` option, will return virtual columns by default in version 0.3.17. Version 0.3.19 omits virtual columns with the same usage pattern.

**Comment by NarHakobyan** [comment]
Yeah, we experiencing the same issue, virtual column stopped working. this is urgent‼️

**Comment by jorenvandeweyer** [comment]
Can you create a PR to fix the issue introduced in #10432 but also keep #10431 fixed? If not, I think we should create a PR reverting the changes from #10432

**Comment by fatlirmorina** [comment]
Same here ->

Upgraded from 0.3.17 to 0.3.19 (latest) and everything related to VirtualColumn breaks.

**Comment by EliezerMafra** [comment]
Same here ->
the `@VirtualColumn` decorator does not work on `0.3.19` version

**Comment by PatrickWe** [comment]
I have the Same issue.

Upgraded from 0.3.17 to 0.3.19 and everything related to VirtualColumn breaks.

**Comment by AmrAnwar** [comment]
not working with us too after update from 0.3.17

**Comment by martin-scholz-emnify** [comment]
still broken in 0.3.20

**Comment by nodegin** [comment]
@gwen1230 I think they are not going to fix this because not much people are using virtual column, so for stability either downgrade or switch to Prisma instead.

**Comment by Dawar** [comment]
> I've seen this too after upgrading. A possible workaround is to pass a select which includes the virtual column, but this isn't great as you then have to pass _all_ the fields that you want to be returned.

Tried this but didn't seem to work, TypeORM is trying to load the column from the actual table. Did you have any success with it? 

If the general direction is to manually select virtualcolumns that would also be fine (there should probably be an eager option on the decorator in that case). The previous virtualcolumn functionality was ideal imo. Curious why these breaking changes were implemented from a benefit stand-point.

Will try and compare prev and curr implementation for possible ideas

**Comment by d3tr1tus** [comment]
Any updates? I would appreciate some fix here it will be really helpful 🙏🏼

**Comment by tristin-terry** [comment]
A fix for this would be helpful, we were just affected by this in production and are going to have to downgrade or patch it internally until it is fixed!

hopefully either https://github.com/typeorm/typeorm/pull/10777 or https://github.com/typeorm/typeorm/pull/10698 can be reviewed soon!

**Comment by fatlirmorina** [comment]
Any news on this, please?

Many production environments are affected by this, please.

**Comment by nodegin** [comment]
@fatlirmorina The solution is to install 0.3.17

**Comment by jsnazaroff** [comment]
Any news on this

**Comment by ReneeDress** [comment]
Any news on this

**Comment by macleysousa** [comment]
> Any news on this

Unfortunately not, still awaiting PR approval.

**Comment by krishnabrq** [comment]
Have you got any update on this yet?

**Comment by Helveg** [comment]
I'd also like to offer my own fork as a solution. I took e7649d2746f907ff36b1efb600402dedd5f5a499 and applied the fix of #10698, and it is published as `@helveg/typeorm`. I won't maintain the fork any further, but people looking for typeorm `0.3.20` with working VirtualColumns can install it with the following npm-alias:

```
npm install typeorm@npm:@helveg/typeorm@0.3.20
```

which will fix this issue.

**Comment by luscala** [comment]
+1

**Comment by DarthSnufkin** [comment]
+1

**Comment by EliezerMafra** [comment]
+1

**Comment by alumni** [comment]
Does https://github.com/typeorm/typeorm/pull/10927 fix this issue?

**Comment by scr4bble** [comment]
Seems like there are several PRs which aim to fix this, right? I am not too deep in the issue but I see a lot of attention here and effort from several people to fix this.

PRs:
https://github.com/typeorm/typeorm/pull/10698
https://github.com/typeorm/typeorm/pull/10777
https://github.com/typeorm/typeorm/pull/10927

@pleerock could you please take a look at them (check whether they look good) and comment on what is needed to be done before they can be merged?

**Comment by alumni** [comment]
https://github.com/typeorm/typeorm/pull/10698 and https://github.com/typeorm/typeorm/pull/10777 revert the PR that introduced the issue. 
https://github.com/typeorm/typeorm/pull/10927 reverts the PR that introduced the issue and fixed the original issue.

**Comment by alumni** [comment]
fixed in #10927
